storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -2752,6 +2890,30 @@ static bool xtrabackup_write_info(const char *filepath) {
   return result;
 }
 
+std::string ddl_lock_type_to_str(lock_ddl_type_t type) {
+  switch (type) {
+    case LOCK_DDL_ON:
+      return "ON";
+    case LOCK_DDL_OFF:
+      return "OFF";
+    case LOCK_DDL_REDUCED:
+      return "REDUCED";
+    default:
+      ut_error;
+  }
+}
+
+lock_ddl_type_t ddl_lock_type_from_str(std::string type) {
+  if (type == "ON") {
+    return LOCK_DDL_ON;
+  } else if (type == "OFF") {
+    return LOCK_DDL_OFF;
+  } else if (type == "REDUCED") {
+    return LOCK_DDL_REDUCED;
+  } else {
+    ut_error;
+  }
+}
```

Comment:
what if there is TRUE and FALSE. This is allowed too. Please update this.
What about case comparison?

Reply:
AB
I think its better to leave if comparisons because switch type is string